test/helpers/testabilityPatch.js

```diff
@@ -381,6 +381,9 @@ function generateInputCompilerHelper(helper) {
         // Compile the lot and return the input element
         $compile(helper.formElm)(scope);
 
+        $rootElement.append(helper.formElm);
+        jqLite($document[0].body).append($rootElement);
```

Comment:
We recently changed `browserTrigger` to manually raise events if the node is not connected to a document - https://github.com/angular/angular.js/commit/c8bfbfc5d7d66b4913874ae2f1c0df6c0dea7857 - would that solve this problem?